base/linalg/triangular.jl

```diff
@@ -1758,6 +1758,7 @@ function logm(A0::UpperTriangular{T}) where T<:Union{Float64,Complex{Float64}}
 
     # Compute repeated roots
     d = complex(diag(A))
+    dm1 = Vector{T}(n)
```

Comment:
this one isn't necessary, and the one below wouldn't be if you used `dm1 = d .- 1` or `dm1 = d - 1` instead of `dm1 .= d .- 1`

the `.=` is in place, `=` allocates a new vector